src/components/Button/index.tsx

```diff
@@ -385,11 +384,16 @@ export const Button: React.FC<ButtonProps> = React.forwardRef(
         onClick={loading ? () => {} : onClick}
         {...props}
       >
-        {hasAdornment && (
+        {leftAdornment && !loading && (
           <LeftAdornment size={size} isSquare={isSquare}>
             {leftAdornment || <Spinner size={loadingIconSize} />}
```

Comment:
```suggestion
            {leftAdornment}
```

This should be removed